### ISSUE TITLE
ACL not working on modules after upgrading when $root_user variable is used in configuration.php

### ISSUE THREAD

**Issue by pabloarias** [joomla/joomla-cms#16997]
### Steps to reproduce the issue
1. Install Joomla 3.7.3.
2. Add variable $root_user to configuration.php with a username of your site.
3. Publish a module with non public permissions.

### Expected result
The module will be not visible by public users or guest.

### Actual result
The module is shown to public users or guest.

### System information
Joomla 3.7.3
PHP 5.6 or 7

### Additional comments
It seems that the error is in libraries/joomla/access/access.php line 1074:
```
if ($root_user && ($root_user == $user->username || $root_user == $user->id))
```
I think we are trying to check $root_user against a string and an int

**Comment by wojsmol** [comment]
@izharaazmi Plese look on this as you wrote this chenge.

**Comment by brianteeman** [comment]
Root user is only intended to be used in order to reset super user access. It should NEVER be left enabled after super user access has been regained.

**Comment by pabloarias** [comment]
I agree, it should never be left enabled but, during the process, private content shouldn't be public.

**Comment by Artwebco** [comment]
I am faced with the same issue, is there any solution?

**Comment by ggppdk** [comment]
There is a string with int comparison in the code

"somename" == 0
will return true,
because "somename" typecasted to integer is 0 , thus you get
0 == 0

I have made a PR

**Comment by franz-wohlkoenig** [comment]
closed as having PR #17011<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16997">issues.joomla.org/tracker/joomla-cms/16997</a>.</sub>
